src/goto-programs/builtin_functions.cpp

```diff
@@ -664,9 +626,8 @@ void goto_convertt::do_java_new_array(
   t_n->code=code_assignt(lhs, malloc_expr);
   t_n->source_location=location;
 
-  assert(ns.follow(object_type).id()==ID_struct);
```

Comment:
How come this assertion is no longer required.

Reply:
This is already checked by `to_struct_type` on the next line

Reply:
and also line 613

Reply:
I find it is sometimes useful to pull the check out to allow you to document in this specific case why it must be a struct, but this is a preference.